tycho-compiler-plugin/src/main/java/org/eclipse/tycho/compiler/AbstractOsgiCompilerMojo.java

```diff
@@ -581,7 +582,7 @@ public List<String> getClasspathElements() throws MojoExecutionException {
                     .filter(AbstractOsgiCompilerMojo::isValidLocation).distinct();
             classpathLocations.forEach(location -> {
                 String path = location.getAbsolutePath();
-                String entry = path + toString(cpe.getAccessRules());
+                String entry = path + toString(cpe.getAccessRules(), useAccessRules);
                 if (seen.add(entry)) {
```

Comment:
For all but the first entry `seen.add(entry)` will be false. The included paths and the classpath will miss entries.

Reply:
> For all but the first entry `seen.add(entry)` will be false. The included paths and the classpath will miss entries. (with javac that is)

I'm not sure I understand what you want to suggest here, the whole "see" part is that a path could apear multiple times (e.g. with different environments) so it is to make the list not contain it more than once ...